app/src/main/java/com/duckduckgo/app/trackerdetection/TdsClient.kt

```diff
@@ -49,7 +59,12 @@ class TdsClient(
         documentUrl: Uri,
         requestHeaders: Map<String, String>,
     ): Client.Result {
-        val tracker = trackers.firstOrNull { sameOrSubdomain(url, it.domain) } ?: return Client.Result(matches = false, isATracker = false)
+        val tracker = if (optimizeTrackerEvaluation) {
+            val domain = url.host?.let { Domain(it) }
+            trackers.firstOrNull { sameOrSubdomain(domain, it.domain) } ?: return Client.Result(matches = false, isATracker = false)
+        } else {
+            trackers.firstOrNull { sameOrSubdomain(url, it.domain.value) } ?: return Client.Result(matches = false, isATracker = false)
```

Comment:
This small change doesn't affect performance. Domain is an inline value class, and therefore at runtime, instances of the inline class will be represented using the underlying type (String)